tests/smartcam/modules/test_barkdetection.py

```diff
@@ -0,0 +1,45 @@
+"""Tests for smartcam bark detection module."""
```

Comment:
Maybe we should also parameterize the tests to avoid code duplication for these similar modules? Just an idea 🤔

Reply:
reverted to multiple SmartCamModule with repeated code, it's the only version that passes tests.

I've tried parametrize_combine but I cannot get it to work; same end also using @pytest.mark.parametrize .

Reply:
Sorry for the late response, but let's rather fix the tests to avoid code duplication to keep it more maintainable.
If you want to bring your generalized revision back to this PR, I will try to find some time soon to look into fixing the tests.

Reply:
hi @rytilahti , sorry for this late response, I've pushed the branch with `detectionmodule` common class, all the detections inherits from it.

I've tried some test changes but I've no idea actually on how to fix it, thanks.

Reply:
Hi @rytilahti , I’ve found a way to have test passed with no code duplication. Please review this PR, thanks.